pkg/security/registry/podsecuritypolicyreview/rest.go

```diff
@@ -0,0 +1,130 @@
+	if !ok {
+		return nil, kapierrors.NewBadRequest(fmt.Sprintf("not a PodSecurityPolicyReview: %#v", obj))
+	}
+	if errs := securityvalidation.ValidatePodSecurityPolicyReview(pspr); len(errs) > 0 {
+		return nil, kapierrors.NewInvalid(securityapi.Kind(pspr.Kind), "", errs)
+	}
+	ns, ok := kapi.NamespaceFrom(ctx)
+	if !ok {
+		return nil, kapierrors.NewBadRequest("namespace parameter required.")
+	}
+	serviceAccounts, err := getServiceAccounts(pspr.Spec, r.client, ns)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(serviceAccounts) == 0 {
+		glog.Errorf("No service accounts for namespace %s", ns)
+		return nil, kapierrors.NewBadRequest(fmt.Sprintf("unable to find ServiceAccount for namespace: %s", ns))
+	}
+
+	errs := []error{}
+	newStatus := securityapi.PodSecurityPolicyReviewStatus{}
+	for _, sa := range serviceAccounts {
+		userInfo := serviceaccount.UserInfo(ns, sa.Name, "")
+		saConstraints, err := r.sccMatcher.FindApplicableSCCs(userInfo)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("unable to find SecurityContextConstraints for ServiceAccount %s: %v", sa.Name, err))
+			continue
+		}
+		oscc.DeduplicateSecurityContextConstraints(saConstraints)
+		sort.Sort(oscc.ByPriority(saConstraints))
+		var namespace *kapi.Namespace
+		for _, constraint := range saConstraints {
+			var (
+				provider kscc.SecurityContextConstraintsProvider
+				err      error
+			)
+			pspsrs := securityapi.PodSecurityPolicySubjectReviewStatus{}
+			if provider, namespace, err = oscc.CreateProviderFromConstraint(ns, namespace, constraint, r.client); err != nil {
+				errs = append(errs, fmt.Errorf("unable to create provider for service account %s: %v", sa.Name, err))
+				continue
+			}
+			_, err = podsecuritypolicysubjectreview.FillPodSecurityPolicySubjectReviewStatus(&pspsrs, provider, pspr.Spec.Template.Spec, constraint)
+			if err != nil {
+				glog.Errorf("unable to fill PodSecurityPolicyReviewStatus from constraint %v", err)
+				continue
+			}
+			sapsprs := securityapi.ServiceAccountPodSecurityPolicyReviewStatus{pspsrs, sa.Name}
+			newStatus.AllowedServiceAccounts = append(newStatus.AllowedServiceAccounts, sapsprs)
+		}
+
+	}
+	if len(errs) > 0 {
+		return nil, kerrors.NewAggregate(errs)
+	}
+	pspr.Status = newStatus
+	return pspr, nil
+}
+
+func getServiceAccounts(psprSpec securityapi.PodSecurityPolicyReviewSpec, client clientset.Interface, namespace string) ([]*kapi.ServiceAccount, error) {
+	serviceAccounts := []*kapi.ServiceAccount{}
+	//  TODO: express 'all service accounts'
+	//if serviceAccountList, err := client.Core().ServiceAccounts(namespace).List(kapi.ListOptions{}); err == nil {
+	//	serviceAccounts = serviceAccountList.Items
+	//	return serviceAccounts, fmt.Errorf("unable to retrieve service accounts: %v", err)
+	//}
+
+	if len(psprSpec.ServiceAccountNames) > 0 {
+		errs := []error{}
+		for _, saName := range psprSpec.ServiceAccountNames {
+			sa, err := client.Core().ServiceAccounts(namespace).Get(saName)
```

Comment:
If its not already in a shared informer factory, then an issue with a link to here will do.